beacon_node/lighthouse_network/tests/common.rs

```diff
@@ -132,14 +138,28 @@ pub async fn build_node_pair(
     log: &slog::Logger,
     fork_name: ForkName,
     spec: &ChainSpec,
+    protocol: Protocol,
 ) -> (Libp2pInstance, Libp2pInstance) {
     let sender_log = log.new(o!("who" => "sender"));
     let receiver_log = log.new(o!("who" => "receiver"));
 
     let mut sender = build_libp2p_instance(rt.clone(), vec![], sender_log, fork_name, spec).await;
     let mut receiver = build_libp2p_instance(rt, vec![], receiver_log, fork_name, spec).await;
 
-    let receiver_multiaddr = receiver.local_enr().multiaddr()[1].clone();
+    let receiver_multiaddr = match protocol {
+        Protocol::Tcp => receiver
+            .local_enr()
+            .multiaddr_tcp()
+            .first()
+            .unwrap()
+            .clone(),
+        Protocol::Quic => receiver
+            .local_enr()
+            .multiaddr_quic()
+            .first()
+            .unwrap()
+            .clone(),
```

Comment:
barely a nit, but being these tests, why not just `pop` which returns by value instead of by reference?

```suggestion
            .pop()
            .unwrap(),
        Protocol::Quic => receiver
            .local_enr()
            .multiaddr_quic()
            .pop()
            .unwrap(),
```

Reply:
makes sense, thanks. Updated